### PR TITLE
Add track_total_hits support in mvt API

### LINKED ISSUE #77222: Support track_total_hits for vector tile requests
the `track_total_hits` parameter is not supported for mvt requests. 

this would be helpful to accurately track whether the tile-contents are complete. the `gte`-relation leaves some ambiguity, and possibility for off-by-one error.

e.g.
```
GET /kibana_sample_data_logs/_mvt/geo.coordinates/0/0/0
{
  "size": 10000,
  "extent": 4096,
}
```

may give a tile with a `10000`-hitcount and `gte`-relation in the meta layer. Presumably the tile is not complete, but it _could_ be.

```
GET /kibana_sample_data_logs/_mvt/geo.coordinates/0/0/0
{
  "size": 10000,
  "track_total_hits: 100001
}
```

This would unambiguously return to the user that the hit-count is greater-than-or-equal to `100001`, allowing to determine that the tile does not contain all features.

cc @iverase

### PR DESCRIPTION
**Pull request by iverase**
The PR adds support for `track_total_hits` in the MVT API. This allows consumers of the API to be able to know exactly if all the features in a tile has been considered when build the hits layer of a vector tile. Implementation is equivalent to the one in the search API.

closes https://github.com/elastic/elasticsearch/issues/77222

### CODE REVIEW

File: x-pack/plugin/vector-tile/src/yamlRestTest/resources/rest-api-spec/test/10_basic.yml [hunk 1]
```diff
@@ -167,3 +167,27 @@ setup:
         zoom: 0
         body:
           sort: []
+
```
Comment (sethmlarson) [inline x-pack/plugin/vector-tile/src/yamlRestTest/resources/rest-api-spec/test/10_basic.yml:170]: Thanks for adding test cases for `track_total_hits` in the request body as well.

**Review by sethmlarson — APPROVED** [review]
LGTM from a lang clients perspective.

**Review by jrodewig — APPROVED** [review]
I pushed 35f9fabd84ff0a1798069b8650bb525a4ba88960 to reuse `track_total_hits` in the request body parameters.

With that change, these docs LGTM.

**Review by imotov — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**